### DISCUSSION TITLE
Was Langium's features made by Langium?

### DISCUSSION THREAD

**Discussion by negahama** [eclipse-langium/langium#1644] [Q&A] [answered]
It's just my curiosity
As more as I learned Langium, Langium is perfect tools for making the Langium's features.
long time ago, I heard C compiler is re-written by C language after C compiler was made...
Was Langium re-written by Langium? ^^

**Comment by msujew** [comment, marked as answer]
Hey @negahama,

Yes, Langium's grammar language is completely self-hosted. You will find it's implementation [here](https://github.com/eclipse-langium/langium/tree/main/packages/langium/src/grammar).

The first few commits in this repo featured a hand-written parser, that was replaced pretty quickly with a Langium-based implementation. You're completely right in that regard :)

**Reply by negahama** [reply]
It's cool...

**Reply by spoenemann** [reply]
See https://en.wikipedia.org/wiki/Bootstrapping_(compilers)
